tests/test_button_block.py

```
@@ -0,0 +1,218 @@
+        """Test ButtonBlock when href is omitted (should fallback to '#')."""
+        self._run_button_block_test("button_no_href_fallback")
+
+    def test_button_all_params_link(self):
+        """Test a link button with many parameters set."""
+        self._run_button_block_test("button_all_params_link")
+        
+    def test_button_different_types_and_sizes(self):
+        """Test various combinations of types and sizes (can be one MD file with multiple buttons)."""
+        self._run_button_block_test("button_various_types_sizes")
+    
+    def test_button_different_types_and_sizes(self):
```

Ruff (0.11.9)

Redefinition of unused `test_button_different_types_and_sizes` from line 203

(F811)